procdiscovery/pkg/process/process.go

```diff
@@ -121,6 +123,22 @@ func (pcx *ProcessContext) GetMapsFile() (ProcessFile, error) {
 	return pcx.mapsFile, nil
 }
 
+// GetOriginalExeFile tries to open the real application binary. When Dynatrace
+// OneAgent is used for Go applications the /proc/<pid>/exe symlink points to
+// oneagentdynamizer. In this case the command line contains the actual binary
+// path as the next argument. This method resolves the path and opens the
+// original binary so language detection can read build information.
+func (pcx *ProcessContext) GetOriginalExeFile() (ProcessFile, error) {
+	exePath := pcx.ExePath
+	if strings.Contains(filepath.Base(exePath), "oneagentdynamizer") {
+		args := strings.Split(pcx.CmdLine, "\x00")
+		if len(args) > 1 && args[1] != "" {
```

Comment:
what do these 2 lins do? what's that split character, and what's the condition where the arg length/values aren't what we're checking here?